docs/shadow.md

```diff
@@ -30,6 +30,31 @@ This uses custom elements, `my-app`, `recipe-hello`, `recipe-hello-binding` and
 
 ## Locate Shadow Dom
 
+### Option 1: Use the [query-selector-shadow-dom](https://github.com/Georgegriff/query-selector-shadow-dom) plugin
+
+CodeceptJS provides an option to apply the `customLocatorStrategy` to locate the Shadow Doms at ease. The query-selector-shadow-dom plugin pierce Shadow DOM roots without knowing the path through nested shadow roots. To use the plugin, define the `customLocatorStrategy` as shown below, and use the `custom` type locators in the tests,
+
+```js
+// in codecept.conf.js
+const { locatorStrategy } = require('query-selector-shadow-dom/plugins/webdriverio');
+
+// under WebDriver Helpers Configuration
+WebDriver: {
+  ...
+  customLocatorStrategy: locatorStrategy
+}
+```
+
+To locate the Shadow DOM in above HTML,
+
+```js
+I.click({custom: '.primary-button'});
```

Comment:
Does this implementation only allow a single custom location strategy?

I wonder if it might be possible to have something like

```js
WebDriver: {
  ...
  customLocatorStrategies: {
      shadowDom: locatorStrategy,
      otherCustomStrat: foo
   }   
}
``
Then 

```js
I.click({shadowDom: '.primary-button'});
// or
I.click({otherCustomStrat: "button"});
```

Reply:
It currently does not but great thing to have. I can update to have the `{}` at least with `shadowDom` instead of just saying `custom` if that works.

Reply:
I hope you don't mind but ive implemented a first pass at how we could support more than one custom locator. i'll open pr into your fork for us to discuss

Reply:
@Georgegriff - that's great! Looking forward to your PR

Reply:
Just writing tests and ill send over :D 
![image](https://user-images.githubusercontent.com/9056958/106955637-6758e380-672d-11eb-83b7-60a67369f5cf.png)

Reply:
@gkushang heres the pr https://github.com/gkushang/CodeceptJS/pull/1

Reply:
@Georgegriff it appears that these tests `customLocatorStrategies` are failing in the Travis CI - https://travis-ci.org/github/codeceptjs/CodeceptJS/jobs/763593653. Would you mind to take a look please?

Reply:
It actually passed for me locally. I am not sure why it failed in TravisCI. Or if TravisCI has some cache? Do you have any idea? @Georgegriff 

<img width="666" alt="Screen Shot 2021-03-19 at 4 26 05 PM" src="https://user-images.githubusercontent.com/3663389/111838814-3e9d3f80-88d0-11eb-8702-918d39cd23a8.png">

Reply:
I fixed the tests and they are passing now ✅